pool/store/mariadb_store.py

```diff
@@ -39,15 +39,17 @@ class MariadbPoolStore(AbstractPoolStore):
     Pool store based on MariaDB.
     """
     async def connect(self):
-        self.pool = await aiomysql.create_pool(
-        minsize=1, 
-        maxsize=12,
-        host=db_host,
-        port=db_port,
-        user=db_user,
-        password=db_password,
-        db=db_name )
-        
+        try:
+            self.pool = await aiomysql.create_pool(
+            minsize=1, 
+            maxsize=12,
+            host=db_host,
+            port=db_port,
+            user=db_user,
+            password=db_password,
+            db=db_name )
+        except Exception as e:
```

Comment:
you should not be catching `Exception` here, but those exceptions which actually indicate a bad config. Specifically ctrl+c would cause InterruptError but that would be caught here and would crash the program indicating a bad config, which would be very confusing. Other errors in the code which initializes the connector such as DNS issues would end up saying "bad config" if you catch too many errors.

Reply:
Fixed by being more specific by using  aiomysql.Error.InterfaceError which is outlined in PEP 249